library/alloc/src/fmt.rs

```diff
@@ -604,9 +604,14 @@ use crate::string;
 #[cfg(not(no_global_oom_handling))]
 #[must_use]
 #[stable(feature = "rust1", since = "1.0.0")]
+#[inline]
 pub fn format(args: Arguments<'_>) -> string::String {
-    let capacity = args.estimated_capacity();
-    let mut output = string::String::with_capacity(capacity);
-    output.write_fmt(args).expect("a formatting trait implementation returned an error");
-    output
+    fn format_cold(args: Arguments<'_>) -> string::String {
```

Comment:
It might be worth commenting that this shouldn't *actually* be `#[cold]`, just non-inline, in order to avoid pessimizing callers (since IIUC blocks dominated by a call to a `#[cold]` function are themselves considered cold).

E.g. this does it right, but might look wrong given the name and lack of attribute.

Reply:
Maybe a `format_inner` then?

Reply:
Sure